code/modules/mob/living/simple_animal/posessed_object.dm

```diff
@@ -155,3 +169,22 @@
 	overlays = possessed_item.overlays
 	set_opacity(possessed_item.opacity)
 	return ..(NONE)
+
+/mob/living/simple_animal/possessed_object/attackby(obj/item/O, mob/living/user)
+	. = ..()
+	if(istype(O, /obj/item/nullrod))
+		visible_message("<span type='notice'>[O] dispels the spooky aura!</span>")
+		death(0)
+
+/mob/living/simple_animal/possessed_object/throw_impact(atom/hit_atom, throwingdatum)
+	. = ..()
+	if(isliving(hit_atom))
```

Comment:
Isn't it better to call `possessed_item.throw_impact(hit_atom, throwingdatum)`? It *should* behave as you want. You'd have to check ofc.
Be sure to check if the possessed item still exists after the throw_impact. Bottles can break for instance. Also, embeds can happen, this might be something you can use to do some fun stuff ;)

Reply:
Do you mean in `/obj/effect/proc_holder/spell/aoe/revenant/haunt_object/proc/attack`? If so I don't think so as the projectile should be actually thrown at the target and only do damage if it hits, as opposed to directly calling `throw_impact`

Reply:
Oh no I meant here. If the possessed object is thrown against something then this proc is called. So instead of having special handling here you'd call `possessed_item.throw_impact(...)` if the `hit_atom` is a living mob.
This would make throwing a possessed item mob behave the same as throwing the item it possesses.
Though I can understand it if you want to have custom behaviour here

Reply:
> Oh no I meant here. If the possessed object is thrown against something then this proc is called. So instead of having special handling here you'd call `possessed_item.throw_impact(...)` if the `hit_atom` is a living mob. This would make throwing a possessed item mob behave the same as throwing the item it possesses.
> 
> Though I can understand it if you want to have custom behaviour here. So the code as is will also work in that case

OH! That could actually work yes, it would mean tweaking throwforce back after possession in the case of revs but I can't see why that would be a problem. It would also mean non-mobs get it as normal too if they get in the way.